### PR TITLE
ENH: print DataFrame columns names at the end

### LINKED ISSUE #32296: [FR] Add keyword or option to print column names at bottom of output, not just top.
```python
import numpy as np
import pandas as pd
pd.DataFrame(np.random.random((4,3)), columns=['a','b','c'])
```

Generates 

```
|   |        a |         b |        c |
|---+----------+-----------+----------|
| 0 |  0.87267 |  0.429153 | 0.545497 |
| 1 | 0.596757 |  0.811766 | 0.797137 |
| 2 | 0.175432 | 0.0254871 | 0.560906 |
| 3 | 0.426507 |   0.39993 | 0.418244 |
```

#### Problem description

Org-mode or Jupyter Notebooks often include Pandas table output. Pandas does try to intelligently limit printed output, but there are times when it is useful to have tables larger than can fit on screen. In this case, having column names repeated at the bottom (or even every `repeat` row) would improve readability and usability.

#### Expected Output

If set to print at bottom, it would be nice to have the option to generate this:

```
|   |        a |         b |        c |
|---+----------+-----------+----------|
| 0 |  0.87267 |  0.429153 | 0.545497 |
| 1 | 0.596757 |  0.811766 | 0.797137 |
| 2 | 0.175432 | 0.0254871 | 0.560906 |
| 3 | 0.426507 |   0.39993 | 0.418244 |
|---+----------+-----------+----------|
|   |        a |         b |        c |
```

If set to print headers every 2nd row, then it might look like this:

```
|   |        a |         b |        c |
|---+----------+-----------+----------|
| 0 |  0.87267 |  0.429153 | 0.545497 |
| 1 | 0.596757 |  0.811766 | 0.797137 |
|---+----------+-----------+----------|
|   |        a |         b |        c |
|---+----------+-----------+----------|
| 2 | 0.175432 | 0.0254871 | 0.560906 |
| 3 | 0.426507 |   0.39993 | 0.418244 |
|---+----------+-----------+----------|
|   |        a |         b |        c |
```

### PR DESCRIPTION
**Pull request by JDkuba**
- [ ] closes #32296
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I've added the possibility of printing columns names again at the end of DataFrame. I'm not sure whether the number of rows should be checked again (max number of rows could be exceeded with that extra row).

### CODE REVIEW

File: pandas/tests/io/formats/test_format.py [hunk 1]
```diff
@@ -2137,6 +2137,13 @@ def test_period(self):
         )
         assert str(df) == exp
 
+    def test_to_string_ending_header(self):
+        tm.reset_display_options()
```
Comment (MarcoGorelli) [inline pandas/tests/io/formats/test_format.py:2141]: Add the issue number here as a comment

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Looks good, can you add a whatsnew note too?

File: pandas/tests/io/formats/test_format.py (hunk 1 above)
Comment (JDkuba) [reply pandas/tests/io/formats/test_format.py:2142]: Done

**Comment by JDkuba** [comment]
@MarcoGorelli Thanks for your comments. I wasn't sure if `ending_header` is the best name for such a parameter. Is it fine?
I've added whatsnew entry, I hope that it is enough descriptive. Please let me know if it should be improved.

File: pandas/io/formats/format.py [hunk 2]
```diff
@@ -133,6 +133,8 @@
             Display DataFrame dimensions (number of rows by number of columns).
         decimal : str, default '.'
             Character recognized as decimal separator, e.g. ',' in Europe.
+        ending_header : bool, optional
+            Display header at the end of DataFrame.
```
Comment (MarcoGorelli) [inline pandas/io/formats/format.py:137]: Can you mention what the default behaviour is?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Great, thanks - I think `ending_header` makes sense, just got a comment about the docstring (which I should've made last time, sorry)

File: pandas/io/formats/format.py (hunk 2 above)
Comment (JDkuba) [reply pandas/io/formats/format.py]: Done

**Comment by WillAyd** [comment]
This looks good - can you merge master? I think should resolve the CI issue

**Comment by JDkuba** [comment]
> This looks good - can you merge master? I think should resolve the CI issue

@WillAyd Done

**Comment by WillAyd** [comment]
@jreback thoughts here?

**Review by jreback — CHANGES_REQUESTED** [review]
this PR is only adding to to_string() which doesn't appear to be very useful, the default formatters won't have this and now you have confusing. I am ok with this in principle but would need to comprehensively audit the display outputs.

cc @jorisvandenbossche @TomAugspurger

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -133,6 +133,8 @@
             Display DataFrame dimensions (number of rows by number of columns).
         decimal : str, default '.'
             Character recognized as decimal separator, e.g. ',' in Europe.
+        ending_header : bool, optional, default False
+            Display header at the end of DataFrame.
```
Comment (jreback) [inline pandas/io/formats/format.py:137]: can you add a versionadded 1.1 tag here

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment, pls ping on green

**Comment by MarcoGorelli** [comment]
Hi @JDkuba - is this still active?

File: pandas/io/formats/format.py (hunk 3 above)
Comment (jreback) [reply pandas/io/formats/format.py:137]: can you do this here

**Review by jreback — CHANGES_REQUESTED** [review]
@JDkuba just a small comment, pls merge master and ping on green.

**Comment by jreback** [comment]
@JDkuba can you merge master and update for comments

**Comment by MarcoGorelli** [comment]
Closing as stale (no response in a month and a half), but please ping if you'd like to bring this back up

**Pull request closed without merging**